jlh/JonFirstApp/app/src/main/java/com/e_conomic/jonfirstapp/MainFragment.java

```diff
@@ -1,26 +1,68 @@
 package com.e_conomic.jonfirstapp;
 
 import android.content.Context;
+import android.content.Intent;
+import android.database.Cursor;
+import android.net.Uri;
 import android.os.Bundle;
+import android.provider.ContactsContract;
 import android.support.annotation.Nullable;
 import android.support.v4.app.Fragment;
+import android.support.v4.app.LoaderManager;
+import android.support.v4.content.CursorLoader;
+import android.support.v4.content.Loader;
+import android.telephony.SmsManager;
+import android.util.Log;
 import android.view.LayoutInflater;
 import android.view.ViewGroup;
 import android.view.View;
 import android.widget.Button;
 import android.widget.EditText;
+import android.widget.LinearLayout;
+import android.widget.TextView;
 
-public class MainFragment extends Fragment {
+import java.util.HashMap;
+import java.util.Map;
+
+import static android.app.Activity.RESULT_OK;
+
+public class MainFragment extends Fragment implements LoaderManager.LoaderCallbacks<Cursor> {
 
     MainFragmentListener delegate;
 
+    private LoaderManager loaderManager;
+
     private Boolean isDisplayMessageFragmentVisible = true;
+    private Boolean isRecipientAdded = false;
+
+    private String contactPhoneNumber = null;
+    private String contactName = null;
+
+    // Request codes
+    final static int PICK_CONTACT_REQUEST = 1;
+
+    final static String MAIN_FRAGMENT_TAG = "MainFragment";
+    final static String ENTER_MESSAGE_FRAGMENT_TAG = "EnterMessageFragment";
+
+    // Keys
+    final static String PHONE_NUMBER = "contactPhoneNumber";
+    final static String DISPLAY_NAME = "displayName";
+    final static String CONTACT_DETAILS = "phoneValues";
+    final static String PHONE_DIRECTORY_CONTENT_TYPE = "phoneDirectoryContentType";
+
+    // Map with phone values.
+    private Map<String, String> phoneValues = new HashMap<>();
```

Comment:
I'm blanking a bit - is there a better way to do this? Referring to the type of the second parameter: they are all taking from `ContactsContract.CommonDataKinds.Phone`.